packages/core/feature-flags/core-feature-flags-browser-internal/src/feature_flags_service.ts

```diff
@@ -0,0 +1,183 @@
+    };
+  }
+
+  /**
+   * Start lifecycle method
+   */
+  public async start(): Promise<FeatureFlagsStart> {
+    const featureFlagsChanged$ = new Subject<string[]>();
+    this.featureFlagsClient.addHandler(ClientProviderEvents.ConfigurationChanged, (event) => {
+      if (event?.flagsChanged) {
+        featureFlagsChanged$.next(event.flagsChanged);
+      }
+    });
+    const observeFeatureFlag$ = (flagName: string) =>
+      featureFlagsChanged$.pipe(
+        filter((flagNames) => flagNames.includes(flagName)),
+        startWith([flagName]) // only to emit on the first call
+      );
+
+    await this.waitForProviderInitialization();
+
+    return {
+      appendContext: (contextToAppend) => this.appendContext(contextToAppend),
+      getBooleanValue: (flagName: string, fallbackValue: boolean) =>
+        this.evaluateFlag(this.featureFlagsClient.getBooleanValue, flagName, fallbackValue),
+      getStringValue: <Value extends string>(flagName: string, fallbackValue: Value) =>
+        this.evaluateFlag<Value>(this.featureFlagsClient.getStringValue, flagName, fallbackValue),
+      getNumberValue: <Value extends number>(flagName: string, fallbackValue: Value) =>
+        this.evaluateFlag<Value>(this.featureFlagsClient.getNumberValue, flagName, fallbackValue),
+      getBooleanValue$: (flagName, fallbackValue) => {
+        return observeFeatureFlag$(flagName).pipe(
+          map(() =>
+            this.evaluateFlag(this.featureFlagsClient.getBooleanValue, flagName, fallbackValue)
+          )
+        );
+      },
+      getStringValue$: <Value extends string>(flagName: string, fallbackValue: Value) => {
+        return observeFeatureFlag$(flagName).pipe(
+          map(() =>
+            this.evaluateFlag<Value>(
+              this.featureFlagsClient.getStringValue,
+              flagName,
+              fallbackValue
+            )
+          )
+        );
+      },
+      getNumberValue$: <Value extends number>(flagName: string, fallbackValue: Value) => {
+        return observeFeatureFlag$(flagName).pipe(
+          map(() =>
+            this.evaluateFlag<Value>(
+              this.featureFlagsClient.getNumberValue,
+              flagName,
+              fallbackValue
+            )
+          )
+        );
+      },
+    };
+  }
+
+  /**
+   * Stop lifecycle method
+   */
+  public async stop() {
+    await OpenFeature.close();
+  }
+
+  /**
+   * Waits for the provider initialization with a timeout to avoid holding the page load for too long
+   * @private
+   */
+  private async waitForProviderInitialization() {
+    // Adding a timeout here to avoid hanging the start for too long if the provider is unresponsive
+    let timeoutId: NodeJS.Timeout | undefined;
+    await Promise.race([
+      this.isProviderReadyPromise,
+      new Promise((resolve) => {
+        timeoutId = setTimeout(resolve, 2 * 1000);
+      }).then(() => {
+        this.logger.warn('The feature flags provider took too long to initialize');
```

Comment:
nit: maybe worth registering this case in APM? Up to you!

Reply:
Yeah, and also push this timeout event as an error? Really minor, up to you!

Reply:
Added in [`9026d32` (#188562)](https://github.com/elastic/kibana/pull/188562/commits/9026d328539d35c7602197471a67036ed6528478).

I used `apm.captureError` because the RUM agent can only run one transaction/span at a time (this is different in the server). So defining a new transaction would automatically stop the automatic transaction `page-load`. Similarly, creating a new span breaks the `GET /app/*` span.

On top of that, I couldn't find in the RUM APIs a way to claim that a transaction/span ended unsuccessfully (the `.end()` method doesn't accept any arguments.